demystifying_machine_learning/demystifying_machine_learning.md

```diff
@@ -0,0 +1,755 @@
+
+<details>
+
+<summary> Want to see the R code that generated those plots? </summary>
+
+```r
+# generate fake quadratic trend data
+n <- 100
+set.seed(8675309)
+
+# x is randomly sampled from a normal distribution
+x <- rnorm(n=n, mean = 0, sd = 1)
+# y is x squared, plus random noise
+y <- -1*x^2 + rnorm(n=n, mean = 0, sd = 2)
+
+# trend lines
+linear <- predict(lm(y ~ x))
+quadratic <- predict(lm(y ~ poly(x, 2, raw = TRUE)))
+nthpoly <- predict(lm(y ~ poly(x, n-2, raw = TRUE)))
+
+# put it into a data frame for plotting
+df <- data.frame(x=x,
+                 y=y,
+                 l = linear,
+                 q = quadratic,
+                 n = nthpoly)
+
+# plots
+library(ggplot2)
+
+base_plot <- ggplot(df, aes(x=x, y=y)) +
+  geom_point(alpha = .7) +
+  labs(x="Blood Pressure", y="Cognitive Performance") +
+  scale_x_continuous(breaks = NULL) +
+  scale_y_continuous(breaks = NULL) +
+  theme_classic()
+
+# underfit
+base_plot +
+  geom_line(aes(y=l))
+
+# overfit
+base_plot +
+  geom_line(aes(y=n))
+
+# goodfit
+base_plot +
+  geom_line(aes(y=q))
+
+```
+
+</details>
+
+### Training data and test data
+
+One thing that is very typical of machine learning analyses is splitting your data into (at least) two separate pieces: one to **train** your model and another to **test** it.
+
+That means you use only part of your data to build your model, and you reserve the rest to test the model's performance on fresh data.
+
+Testing your model on fresh data gives you a chance to catch an overfitting problem.
+As you train a model, you can always reduce errors by increasing its flexibility (i.e. making it more complex).
+But on the training data you can't tell if the improvement in model fit is mostly because you're reducing error from bias (good) or mostly because you're modeling random error from the idiosyncrasies of these particular data (bad).
+The only way to find out is by running the model on new data, so you can capture the variance.
+Splitting your data into training and test sets to estimate how well the model will generalize to new data is called [cross validation](https://developers.google.com/machine-learning/glossary#cross-validation).
+
+If you've let your model get too complex in your attempt to reduce bias, then when you test that model on fresh data, the error will go up on the fresh data because your variance got too high.
+
+<div class = "important">
+<b style="color: rgb(var(--color-highlight));">Important note</b><br>
+
+If your measure of model fit on the test data is substantially worse than your measure of model fit on the training data, you know you've overfit the model.
+
+</div>
+
+For an excellent video summary of how splitting your data into training and test sets helps with the bias-variance tradeoff, see this StatsQuest video (note the section on splitting your data starts about halfway through, but feel free to also watch the beginning if you want to get a review of the definitions of bias and variance):
```

Comment:
Weird that it's not starting the video at the specified time! Liascript used to handle that correctly, but apparently not anymore. I've updated the text before the video to no longer specifically tell readers to expect it to start at the halfway point; I'm hoping this wording will work now with the video starting at the beginning and also hopefully with it beginning halfway through, if that functionality ever magically reappears.